src/classes/Shapes/CompoundShape.js

```diff
@@ -47,52 +74,78 @@ Goblin.CompoundShape.prototype.calculateLocalAABB = function( aabb ) {
 	}
 };
 
-Goblin.CompoundShape.prototype.getInertiaTensor = function( mass ) {
+Goblin.CompoundShape.prototype.computeSteiner = function ( vector, mass, tensor ) {
+	tensor.e00 = mass * -( vector.y * vector.y + vector.z * vector.z );
+	tensor.e10 = mass * vector.x * vector.y;
+	tensor.e20 = mass * vector.x * vector.z;
+
+	tensor.e01 = mass * vector.x * vector.y;
+	tensor.e11 = mass * -( vector.x * vector.x + vector.z * vector.z );
+	tensor.e21 = mass * vector.y * vector.z;
+
+	tensor.e02 = mass * vector.x * vector.z;
+	tensor.e12 = mass * vector.y * vector.z;
+	tensor.e22 = mass * -( vector.x * vector.x + vector.y * vector.y );
+};
+
+Goblin.CompoundShape.prototype.getInertiaTensor = function( _mass ) {
 	var tensor = new Goblin.Matrix3(),
 		j = new Goblin.Matrix3(),
 		i,
 		child,
 		child_tensor;
 
-	mass /= this.child_shapes.length;
+	if ( this.child_shapes.length === 0 ) {
+		return tensor;
+	}
+
+	var mass = _mass / this.child_shapes.length;
 
-	// Holds center of current tensor
-	_tmp_vec3_1.x = _tmp_vec3_1.y = _tmp_vec3_1.z = 0;
+	// our origin is current center
+	_tmp_vec3_1.copy( this.center_of_mass );
 
 	for ( i = 0; i < this.child_shapes.length; i++ ) {
 		child = this.child_shapes[i];
 
-		_tmp_vec3_1.subtract( child.position );
-
-		j.e00 = mass * -( _tmp_vec3_1.y * _tmp_vec3_1.y + _tmp_vec3_1.z * _tmp_vec3_1.z );
-		j.e10 = mass * _tmp_vec3_1.x * _tmp_vec3_1.y;
-		j.e20 = mass * _tmp_vec3_1.x * _tmp_vec3_1.z;
-
-		j.e01 = mass * _tmp_vec3_1.x * _tmp_vec3_1.y;
-		j.e11 = mass * -( _tmp_vec3_1.x * _tmp_vec3_1.x + _tmp_vec3_1.z * _tmp_vec3_1.z );
-		j.e21 = mass * _tmp_vec3_1.y * _tmp_vec3_1.z;
-
-		j.e02 = mass * _tmp_vec3_1.x * _tmp_vec3_1.z;
-		j.e12 = mass * _tmp_vec3_1.y * _tmp_vec3_1.z;
-		j.e22 = mass * -( _tmp_vec3_1.x * _tmp_vec3_1.x + _tmp_vec3_1.y * _tmp_vec3_1.y );
+		_tmp_vec3_1.subtract( child.local_position );
+		this.computeSteiner( _tmp_vec3_1, mass, j );
```

Comment:
Last computeSteiner parameter is tensor, but j is passed, is it intentional?

Reply:
It's just named tensor for readability in the function; j is a local variable of the method using it.